File: idscp2/src/main/java/de/fhg/aisec/ids/idscp2/idscp_core/secure_channel/SecureChannelListener.java
package de.fhg.aisec.ids.idscp2.idscp_core.secure_channel;

/**
 * An interface for a secure channel listener, implemented by the secure channel
 */
public interface SecureChannelListener {

    /*
     * Delegate data from secure channel endpoint to the secure channel
     */
    void onMessage(byte[] data);

    /*
     * Delegate an error from an secure channel endpoint to the secure channel
     */
    void onError(Throwable t);

    /*
     * Notify secure channel that secure channel endpoint has been closed
     */
    void onClose();
}

File: idscp2/src/main/kotlin/de/fhg/aisec/ids/idscp2/idscp_core/secure_channel/SecureChannelListener.kt
package de.fhg.aisec.ids.idscp2.idscp_core.secure_channel

/**
 * An interface for a secure channel listener, implemented by the secure channel
 */
interface SecureChannelListener {
    /*
     * Delegate data from secure channel endpoint to the secure channel
     */
    fun onMessage(data: ByteArray)

    /*
     * Delegate an error from an secure channel endpoint to the secure channel
     */
    fun onError(t: Throwable)

    /*
     * Notify secure channel that secure channel endpoint has been closed
     */
    fun onClose()
}